### ISSUE TITLE
Allow `exported` to be set in config

### ISSUE THREAD

**Issue by tschaub** [eslint/eslint#4899]
In #1200, support was added for `exported` in a comment, but it looks like it is not possible to apply this in a config file.

I'm linting a large directory of examples that all use syntax similar to this:

```js
var demo = new Demo();
```

Rather than adding a `/* exported demo */` comment to hundreds of files, I'd prefer to add a single `.eslintrc` to the directory that allowed the `no-unused-vars` rule to be applied.

Would there be support for adding `exported` in config files?  Alternatively, would it be more appropriate to add an `except` (or similar) option to the `no-unused-vars` rule?

**Comment by eslintbot** [comment]
@tschaub Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
Why not add `Demo` as a `global` in `.eslintrc` file?

**Comment by tschaub** [comment]
With this `test-config.json`:
```json
{
  "global": {
    "demo": false
  },
  "rules": {
    "no-unused-vars": 2
  }
}
```

And this `test-script.js`:
```js
var demo = new Demo();
```

I get this:
```
  1:5  error  'demo' is defined but never used  no-unused-vars
```

**Comment by tschaub** [comment]
I had completely missed the `varsIgnorePattern` option.  This fits my use case well (and seems more appropriate than a top level `exported` config property).
